exonum-java-binding/core/src/main/java/com/exonum/binding/core/runtime/ServiceRuntime.java

```diff
@@ -208,37 +209,70 @@ public void initiateAddingService(Fork fork, ServiceInstanceSpec instanceSpec,
   }
 
   /**
-   * TODO(ECR-3919): fix the documentation of the method
-   * Adds a service instance to the runtime after it has been successfully initialized
-   * in {@link #initiateAddingService(Fork, ServiceInstanceSpec, byte[])}. This operation
-   * completes the service instance registration, allowing subsequent operations on it:
+   * Modifies the state of the given service instance at the runtime either by activation it or
+   * stopping. The service instance should be successfully initialized
+   * by {@link #initiateAddingService(Fork, ServiceInstanceSpec, byte[])} in advance.
+   * Activation leads to the service instance registration, allowing subsequent operations on it:
    * transactions, API requests.
+   * Stopping leads to the service disabling i.e. stopped service does not execute transactions,
+   * process events, provide APIs, etc. And it becomes unavailable to other services,
+   * but still exists.
    *
    * @param instanceSpec a service instance specification; must reference a deployed artifact
    * @param instanceStatus a new status of the service instance
-   * @throws IllegalArgumentException if the service is already started; or its artifact
-   *     is not deployed
+   * @throws IllegalArgumentException if activating already active service; or its artifact
+   *     is not deployed; or unrecognized service status received
    */
   public void updateInstanceStatus(ServiceInstanceSpec instanceSpec,
       InstanceState.Status instanceStatus) {
-    try {
-      synchronized (lock) {
-        // Create a previously added service
-        ServiceWrapper service = createService(instanceSpec);
-        // Register it in the runtime
-        registerService(service);
-        // Connect its API
-        connectServiceApi(service);
+    synchronized (lock) {
+      if (instanceStatus.equals(Status.ACTIVE)) {
+        activateService(instanceSpec);
+      } else if (instanceStatus.equals(Status.STOPPED)) {
+        stopService(instanceSpec);
+      } else if (instanceStatus.equals(Status.NONE)) {
+        logger.warn("None status for the service instance {}, ignore. "
```

Comment:
log at info? Also, I am not sure if such questions shall appear in the log (it would be nice if we — as SR developers — do know for sure):
"Possibly restoring services state after reboot?"

What does Runtime spec say?

Reply:
Core Dispatcher invokes updateInstanceStatus for each service in the schema regardless of the status. Not sure why do we restore not active services?

Reply:
Re none — see the reply above; re stopped — are you saying the core confuses a "request to stop a service instance" with a "notification that a service instance **has been stopped already**"? Why is that?

Reply:
> are you saying the core confuses a "request to stop a service instance" with a "notification that a service instance has been stopped already"? Why is that?

@bullet-tooth Have you managed to clarify that? Shall the statuses in core be improved to avoid decoding of what 'stopped' means in each runtime?

Reply:
Created a ticket https://jira.bf.local/browse/ECR-4128